exporter/datadogexporter/translate_traces_test.go

```diff
@@ -282,6 +284,78 @@ func TestTracesTranslationErrorsAndResource(t *testing.T) {
 	assert.Contains(t, datadogPayload.Traces[0].Spans[0].Meta[tagContainersTags], "pod_name:example-pod-name")
 }
 
+func TestTracesTranslationErrorsFromEvents(t *testing.T) {
+	hostname := "testhostname"
+	calculator := newSublayerCalculator()
+	// translate mocks to datadog traces
+	cfg := config.Config{
+		TagsConfig: config.TagsConfig{
+			Version: "v1",
+		},
+	}
+
+	endTime := time.Now().Round(time.Second)
+	pdataEndTime := pdata.TimestampUnixNano(endTime.UnixNano())
+	startTime := endTime.Add(-90 * time.Second)
+	pdataStartTime := pdata.TimestampUnixNano(startTime.UnixNano())
+
+	traces := pdata.NewTraces()
+	traces.ResourceSpans().Resize(1)
+	rs := traces.ResourceSpans().At(0)
+	resource := rs.Resource()
+	resource.Attributes().InitFromMap(map[string]pdata.AttributeValue{
+		"service.name": pdata.NewAttributeValueString("sure"),
+	})
+	rs.InstrumentationLibrarySpans().Resize(1)
+	ilss := rs.InstrumentationLibrarySpans().At(0)
+	instrumentationLibrary := ilss.InstrumentationLibrary()
+	instrumentationLibrary.SetName("flash")
+	instrumentationLibrary.SetVersion("v1")
+
+	ilss.Spans().Resize(1)
+	span := ilss.Spans().At(0)
+	span.Status().SetCode(pdata.StatusCodeError)
+
+	events := span.Events()
+	events.Resize(4)
+
+	events.At(0).SetTimestamp(pdataStartTime)
+	events.At(0).SetName("start")
+
+	events.At(1).SetTimestamp(pdataEndTime)
+	events.At(1).SetName(conventions.AttributeExceptionEventName)
```

Comment:
If possible, would recomend having test cases where the exception is first or last, to find off-by-one errors.

Reply:
👍